Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,221 @@ namespace xr
                     float rawPose[7]{};
                     ArHitResult_getHitPose(xrContext->Session, hitResult, tempPose);
                     ArPose_getPoseRaw(xrContext->Session, tempPose, rawPose);
-                    HitResult hitResult{};
-                    RawToPose(rawPose, hitResult.Pose);
+                    HitResult hitTestResult{};
+                    RawToPose(rawPose, hitTestResult.Pose);
 
-                    hitResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
-                    filteredResults.push_back(hitResult);
+                    hitTestResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
+                    filteredResults.push_back(hitTestResult);
                     frameTrackables.push_back(trackable);
                 }
             }
         }
 
+        // Converts an image bitmap to grayscale assumes an RGB8 or RGB8A image for use with ARCore.
+        void ConvertBitmapToGrayscale(
+            const uint8_t* image_pixel_buffer,
+            int32_t width,
+            int32_t height,
+            int32_t stride,
+            uint8_t** out_grayscale_buffer)
+        {
+            uint8_t* grayscale_buffer = new uint8_t[width * height];
+            uint32_t pixelStride = stride / width;
+            for (int h = 0; h < height; ++h)
+            {
+                for (int w = 0; w < width; ++w)
+                {
+                    const uint8_t* pixel = &image_pixel_buffer[w * pixelStride + h * stride];
+                    uint8_t r = *pixel;
+                    uint8_t g = *(pixel + 1);
+                    uint8_t b = *(pixel + 2);
+                    grayscale_buffer[w + h * width] = static_cast<uint8_t>(0.213f * r + 0.715 * g + 0.072 * b);
+                }
+            }
+            *out_grayscale_buffer = grayscale_buffer;
+        }
+
+        std::vector<std::string> CreateAugmentedImageDatabase(std::vector<System::Session::Frame::ImageTrackingRequest>& requests)
```

Comment:
Looks like the input vector can be const?
```suggestion
        std::vector<std::string> CreateAugmentedImageDatabase(const std::vector<System::Session::Frame::ImageTrackingRequest>& requests)
```